File: 0078-subsets/0078-subsets.go
func subsets(nums []int) [][]int {
    result := [][]int{}
    var backtrack func(cur []int, idx int)

    backtrack = func(cur []int, idx int) {
        if idx == len(nums) {
            result = append(result, cur)
            return
        }
        backtrack(cur, idx+1)
        cur = append(cur, nums[idx])
        curCopy := make([]int, len(cur))
        copy(curCopy, cur)
        backtrack(curCopy, idx+1)        
    }
    backtrack([]int{}, 0)
    return result
}

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        numsCopy = nums;
        vector<int> cur;
        backtrack(0, cur);
        return result;
    }

    void backtrack(int i, vector<int>& cur) {
        if (i == numsCopy.size()) {
            result.push_back(cur);
            return;
        }
        cur.push_back(numsCopy[i]);
        backtrack(i + 1, cur);
        cur.pop_back();
        backtrack(i + 1, cur);
    }
    vector<int> numsCopy;
    vector<vector<int>> result;
};